src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/badge.html.twig

```diff
@@ -24,12 +24,18 @@
  *#}
 
 <div class="text-right">
-{% if record[column.options.field] is not empty %}
+{% if record[column.options.field] is not empty and column.options.badge_type is not empty %}
   <span class="badge rounded badge-{{ column.options.badge_type }}">
     {{ record[column.options.field] }}
   </span>
+{% elseif record[column.options.field] is not empty and column.options.color_field in record|keys and record[column.options.color_field] is not empty %}
+  {% set textColor = is_color_bright(record[column.options.color_field]) ? '#383838' : 'white' %}
```

Comment:
You could use classes here 
{% set textClass = is_color_bright(record[column.options.color_field]) ? 'text-dark' : 'text-bright' %}
and <span class="badge rounded {{ textClass }}" ...>

But as we already have inline style for the background-color, no need :)